xds/internal/balancer/cdsbalancer/cluster_handler.go

```diff
@@ -298,6 +302,19 @@ func (c *clusterNode) handleResp(clusterUpdate xdsresource.ClusterUpdate, err er
 	// Aggregate cluster handling.
 	newChildren := make(map[string]bool)
 	for _, childName := range clusterUpdate.PrioritizedClusterNames {
+		if c.depth == maxDepth-1 {
```

Comment:
why run this inside the loop? Wouldn't it be the same before the loop?

Reply:
No. If the update doesn't have Prioritized Cluster Names, it means that the aggregate cluster doesn't have children, so doesn't exceed max depth. Yes, this adds the overhead of this conditional every iteration through the for loop, but I think it's incorrect to put it outside of the loop.

Reply:
Discussed in QAZSE, moved outside the for loop and into an explicit if.